### PR TITLE
Cameronperera/app token grant type

### PR DESCRIPTION
**Pull request by cameronperera**

### CODE REVIEW

File: internal/apptoken/app_token_grant_test.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+			wantErrContains: "missing app token id",
+			wantErrMatch:    errors.T(errors.InvalidParameter),
+		},
+		{
+			name:            "invalid grant",
+			appTokenId:      testAppTokenId,
+			grant:           "id=*;type=*;actions=",
+			wantErrContains: "parsing grant string",
+			wantErrMatch:    errors.T(errors.InvalidParameter),
+		},
```
Comment (cameronperera) [inline internal/apptoken/app_token_grant_test.go:63]: What should be the correct error to match this against? It is currently passing but I don't think it should be.

Comment (cameronperera) [reply internal/apptoken/app_token_grant_test.go]: Fixed.

**Review by uruemu — APPROVED** [review]

File: internal/apptoken/app_token_grant.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	case appTokenId == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing app token id")
+	case grant == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing grant")
+	}
+
+	// Validate that the grant parses successfully. Note that we fake the scope
+	// here to avoid a lookup as the scope is only relevant at actual ACL
+	// checking time and we just care that it parses correctly.
+	perm, err := perms.Parse(ctx, "o_abcd1234", grant)
```
Comment (jimlambrt) [inline internal/apptoken/app_token_grant.go:36]: perhaps define a const for fakeScopeId to be clear what it is and why

**Review by jimlambrt — APPROVED** [review]
One minor nit, but looks great!

File: internal/apptoken/app_token_grant.go (hunk 2 above)
Comment (cameronperera) [reply internal/apptoken/app_token_grant.go:36]: Good idea. I will update it now.

**Pull request merged**